### PR TITLE
Improvement for matching LLDP neighbors with known hosts.

### PR DESCRIPTION
**Pull request by dagbdagb**
LLDP hosts presenting with FQDN do not currently match with devices stored without a FQDN in the devices table.

This fix also compares LLDP hostname with $hostname + $config['mydomain'], when $config['mydomain'] is set.

This yields recognized neighbors and clickable links in the neighbor overview.

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: includes/discovery/functions.inc.php [hunk 1]
```diff
@@ -1387,6 +1387,9 @@ function find_device_id($name = '', $ip = '', $mac_address = '')
         if ($mydomain = Config::get('mydomain')) {
             $where[] = '`hostname`=?';
             $params[] = "$name.$mydomain";
+
+            $where[] = 'concat(`hostname`, \'.\', \'' . $mydomain . '\') =?';
```
Comment (murrant) [inline includes/discovery/functions.inc.php:1391]: you need to use a ? for $mydomain to avoid sql injection.

Comment (dagbdagb) [reply includes/discovery/functions.inc.php]: Better now?

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/librenms-giving-error-when-trying-to-access-any-page/11881/16

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-64-release-changelog-may-2020/12290/1

**Pull request merged**